### PR TITLE
Add pageblock to customer account UI

### PR DESCRIPTION
**Pull request by kimhawthorne-shopify**
### Background
**TLDR;** We are re-adding `PageBlock`, an exclusive Self-serve component, to the customer-account-ui packages for use in a 1P extension app being built on Self-serve for the [Shopify Returns project](https://vault.shopify.io/projects/23278).

- Add `PageBlock` to both the plain and React flavors of the customer-account-ui packages
- Update the type for the `padding` prop

#### Related PRs
- Original PR to add `PageBlock`: https://github.com/Shopify/ui-extensions/pull/361
- PR to remove `PageBlock`: https://github.com/Shopify/ui-extensions/pull/365

### Solution
In order to unblock the Shopify Returns team, we are re-exposing `PageBlock` as an extension component in customer-account-ui. A different component will likely replace `PageBlock` in the future.

### 🎩

- ...

### Checklist

- [ ] I have :tophat:'d these changes
- [ ] I have updated relevant documentation

### CODE REVIEW

File: packages/customer-account-ui-extensions/src/components/PageBlock/PageBlock.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   *
+   * Examples:
+   *
+   * - `base` means blockStart, inlineEnd, blockEnd and inlineStart paddings are `base`
+   *
+   * - [`base`, `none`] means blockStart and blockEnd paddings are `base`, inlineStart and inlineEnd paddings are `none`
+   *
+   * - [`base`, `none`, `loose`, `tight`] means blockStart padding is `base`, inlineEnd padding is `none`, blockEnd padding is `loose` and  blockStart padding is `tight`
+   */
+  padding?: MaybeConditionalStyle<MaybeShorthandProperty<Spacing>>;
```
Comment (kimhawthorne-shopify) [inline packages/customer-account-ui-extensions/src/components/PageBlock/PageBlock.ts:29]: Originally, this type was:

```
padding?: CSSShorthand<Spacing> | Responsive<CSSShorthand<Spacing>>;
```

However, `CSSShorthand` and `Responsive` has since [been removed](https://github.com/Shopify/ui-extensions/pull/365/files#diff-caeca760aafbe6b1565e10f0fe4dd403ac60588a909237fafe0568ca722cedf6) from checkout-ui-extensions. I updated this type using the newly available types from checkout-ui-extensions:

- `MaybeConditionalStyle`: padding may be conditional, as it [supports conditionals](https://github.com/Shopify/customer-account-web/blob/dc9cd669a3bba9e8a41d8bf6e9ee27b44e19d882/app/components/PageBlock/PageBlock.tsx#L15)
- `MaybeShorthandProperty`: padding can use shorthand that mirrors CSS functionality (see code comment)

**Comment by kimhawthorne-shopify** [comment]
Hi all! 👋 I am new to this repo but am working on a first-party application built on the Self-serve platform. Per [this Slack discussion](https://shopify.slack.com/archives/C0308QJFNQ4/p1657738681436099), we decided to re-expose `PageBlock` to unblock extension app development. 

Please let me know if there's anything I missed here!

File: packages/customer-account-ui-extensions/src/components/PageBlock/PageBlock.ts (hunk 1 above)
Comment (robin-drexler) [reply packages/customer-account-ui-extensions/src/components/PageBlock/PageBlock.ts:29]: Same here. can we put this in a shared file? Like here: https://github.com/Shopify/checkout-web/blob/master/packages/checkout-ui-extensions/src/components/shared.ts#L251

**Review by mathieulag — APPROVED** [review]
🚀
